specreduce/extract.py

```diff
@@ -497,50 +649,79 @@ def __call__(self, image=None, trace_object=None,
             img = _align_along_trace(
                 img,
                 trace_object.trace,
                 disp_axis=disp_axis,
-                crossdisp_axis=crossdisp_axis
-            )
-            # Choose the initial guess for the mean of
-            # the Gaussian profile:
-            mean_init_guess = np.broadcast_to(
-                img.shape[crossdisp_axis] // 2, img.shape[disp_axis]
-            )
-
-        # co-add signal in each image column
-        nrows = img.shape[crossdisp_axis]
-        xd_pixels = np.arange(nrows)  # counted in y dir on plot (or x in spec)
-
-        row_mask = np.logical_or.reduce(or_mask, axis=disp_axis)
-        coadd = np.ma.masked_array(np.sum(img, axis=disp_axis) / nrows,
-                                   mask=row_mask)
-        # (mask rows with non-finite sums for fit to work later on)
-
-        # fit source profile to brightest row, using Gaussian model as template
-        # NOTE: could add argument for users to provide their own model
-        gauss_prof = models.Gaussian1D(amplitude=coadd.max(),
-                                       mean=coadd.argmax(), stddev=2)
-
-        # Fit extraction kernel to column's finite values with combined model
-        # (must exclude masked indices manually; LevMarLSQFitter does not)
-        ext_prof = gauss_prof + bkgrd_prof
-        fitter = fitting.LevMarLSQFitter()
-        fit_ext_kernel = fitter(ext_prof,
-                                xd_pixels[~row_mask], coadd[~row_mask])
+                crossdisp_axis=crossdisp_axis)
+
+        if self.spatial_profile == 'gaussian':
+
+            # fit profile to average (mean) profile along crossdisp axis
+            fit_ext_kernel = self._fit_gaussian_spatial_profile(img,
+                                                                disp_axis,
+                                                                crossdisp_axis,
+                                                                or_mask,
+                                                                bkgrd_prof)
+
+            # this is just creating an array of the trace to shift the mean
+            # when iterating over each wavelength. this needs to be fixed in the
+            # future to actually account for the trace shape in a non-flat trace
+            # (or possibly omitted all togehter as it might be redundant if
+            # _align_along_trace is correcting this already)
+            if isinstance(trace_object, FlatTrace):
+                mean_init_guess = trace_object.trace
+            else:
+                mean_init_guess = np.broadcast_to(
+                    img.shape[crossdisp_axis] // 2, img.shape[disp_axis]
+                )
+
+        else:  # self_profile
+            # for now, bkgrnd_prof must be None because a compound model can't
+            # be created with a interpolator + model. i think there is a way
+            # around this, but will follow up later
```

Comment:
should we create a follow-up issue for this or is it already being tracked somewhere?

Reply:
there's no follow up issue - i think we should consider removing this option altogether which is why i didn't spend the time to do this here.